File: plugins/dotnet/robocode.dotnet.napi/src/_AdvancedRadiansRobot.cs
using System;
using System.Runtime.CompilerServices;
using IKVM.Attributes;
using robocode.robotinterfaces.peer;

namespace robocode
{
    public class _AdvancedRadiansRobot : Robot
    {
        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable((ushort) 0x2c)]
        internal _AdvancedRadiansRobot()
        {
            GC.KeepAlive(this);
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x25, 0x6b, 0x8e, 0x65})]
        public virtual double getGunHeadingRadians()
        {
            if (base.peer != null)
            {
                return base.peer.getGunHeading();
            }
            uninitializedException();
            return 0f;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x75, 0x6b, 0x8e, 0x65})]
        public virtual double getGunTurnRemainingRadians()
        {
            if (base.peer != null)
            {
                return base.peer.getGunTurnRemaining();
            }
            uninitializedException();
            return 0f;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x9f, 0xbd, 0x6b, 0x8e, 0x65})]
        public virtual double getHeadingRadians()
        {
            if (base.peer != null)
            {
                return base.peer.getBodyHeading();
            }
            uninitializedException();
            return 0f;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x2d, 0x6b, 0x8e, 0x65})]
        public virtual double getRadarHeadingRadians()
        {
            if (base.peer != null)
            {
                return base.peer.getRadarHeading();
            }
            uninitializedException();
            return 0f;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x7d, 0x6b, 0x8e, 0x65})]
        public virtual double getRadarTurnRemainingRadians()
        {
            if (base.peer != null)
            {
                return base.peer.getRadarTurnRemaining();
            }
            uninitializedException();
            return 0f;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {160, 0x45, 0x6b, 0x8e, 0x65})]
        public virtual double getTurnRemainingRadians()
        {
            if (base.peer != null)
            {
                return base.peer.getBodyTurnRemaining();
            }
            uninitializedException();
            return 0f;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x35, 0x6b, 0x98, 0x85})]
        public virtual void setTurnGunLeftRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).setTurnGun(-radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x3d, 0x6b, 0x97, 0x85})]
        public virtual void setTurnGunRightRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).setTurnGun(radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {5, 0x6b, 0x98, 0x85})]
        public virtual void setTurnLeftRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).setTurnBody(-radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x45, 0x6b, 0x98, 0x85})]
        public virtual void setTurnRadarLeftRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).setTurnRadar(-radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x4d, 0x6b, 0x97, 0x85})]
        public virtual void setTurnRadarRightRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).setTurnRadar(radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {13, 0x6b, 0x97, 0x85})]
        public virtual void setTurnRightRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).setTurnBody(radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x55, 0x6b, 0x93, 0x85})]
        public virtual void turnGunLeftRadians(double radians)
        {
            if (base.peer != null)
            {
                base.peer.turnGun(-radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x5d, 0x6b, 0x92, 0x85})]
        public virtual void turnGunRightRadians(double radians)
        {
            if (base.peer != null)
            {
                base.peer.turnGun(radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x15, 0x6b, 0x93, 0x85})]
        public virtual void turnLeftRadians(double radians)
        {
            if (base.peer != null)
            {
                base.peer.turnBody(-radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x65, 0x6b, 0x98, 0x85})]
        public virtual void turnRadarLeftRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).turnRadar(-radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x6d, 0x6b, 0x97, 0x85})]
        public virtual void turnRadarRightRadians(double radians)
        {
            if (base.peer != null)
            {
                ((IAdvancedRobotPeer) base.peer).turnRadar(radians);
            }
            else
            {
                uninitializedException();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x1d, 0x6b, 0x92, 0x85})]
        public virtual void turnRightRadians(double radians)
        {
            if (base.peer != null)
            {
                base.peer.turnBody(radians);
            }
            else
            {
                uninitializedException();
            }
        }
    }
}

File: robocode.api/src/main/java/robocode/_AdvancedRadiansRobot.java
/**
 * Copyright (c) 2001-2020 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package robocode;


import robocode.robotinterfaces.peer.IAdvancedRobotPeer;


/**
 * This class is used by the system as a placeholder for all *Radians calls in
 * {@link AdvancedRobot}. You may refer to this class for documentation only.
 * <p>
 * You should create a {@link AdvancedRobot} instead.
 * <p>
 * (The Radians methods themselves will continue work, however).
 *
 * @see Robot
 * @see JuniorRobot
 * @see AdvancedRobot
 * @see TeamRobot
 * @see RateControlRobot
 *
 * @author Mathew A. Nelson (original)
 * @author Flemming N. Larsen (contributor)
 * @author Pavel Savara (contributor)
 */
public class _AdvancedRadiansRobot extends _AdvancedRobot {

	_AdvancedRadiansRobot() {}

	public double getHeadingRadians() {
		if (peer != null) {
			return peer.getBodyHeading();
		}
		uninitializedException();
		return 0; // never called
	}

	public void setTurnLeftRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).setTurnBody(-radians);
		} else {
			uninitializedException();
		}
	}

	public void setTurnRightRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).setTurnBody(radians);
		} else {
			uninitializedException();
		}
	}

	public void turnLeftRadians(double radians) {
		if (peer != null) {
			peer.turnBody(-radians);
		} else {
			uninitializedException();
		}
	}

	public void turnRightRadians(double radians) {
		if (peer != null) {
			peer.turnBody(radians);
		} else {
			uninitializedException();
		}
	}

	public double getGunHeadingRadians() {
		if (peer != null) {
			return peer.getGunHeading();
		}
		uninitializedException();
		return 0; // never called
	}

	public double getRadarHeadingRadians() {
		if (peer != null) {
			return peer.getRadarHeading();
		}
		uninitializedException();
		return 0; // never called
	}

	public void setTurnGunLeftRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).setTurnGun(-radians);
		} else {
			uninitializedException();
		}
	}

	public void setTurnGunRightRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).setTurnGun(radians);
		} else {
			uninitializedException();
		}
	}

	public void setTurnRadarLeftRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).setTurnRadar(-radians);
		} else {
			uninitializedException();
		}
	}

	public void setTurnRadarRightRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).setTurnRadar(radians);
		} else {
			uninitializedException();
		}
	}

	public void turnGunLeftRadians(double radians) {
		if (peer != null) {
			peer.turnGun(-radians);
		} else {
			uninitializedException();
		}
	}

	public void turnGunRightRadians(double radians) {
		if (peer != null) {
			peer.turnGun(radians);
		} else {
			uninitializedException();
		}
	}

	public void turnRadarLeftRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).turnRadar(-radians);
		} else {
			uninitializedException();
		}
	}

	public void turnRadarRightRadians(double radians) {
		if (peer != null) {
			((IAdvancedRobotPeer) peer).turnRadar(radians);
		} else {
			uninitializedException();
		}
	}

	public double getGunTurnRemainingRadians() {
		if (peer != null) {
			return peer.getGunTurnRemaining();
		}
		uninitializedException();
		return 0; // never called
	}

	public double getRadarTurnRemainingRadians() {
		if (peer != null) {
			return peer.getRadarTurnRemaining();
		}
		uninitializedException();
		return 0; // never called
	}

	public double getTurnRemainingRadians() {
		if (peer != null) {
			return peer.getBodyTurnRemaining();
		}
		uninitializedException();
		return 0; // never called
	}
}
